test/reduce.js

```diff
@@ -0,0 +1,385 @@
+var U = require("..");
```

Comment:
Use `"./node"` here, then add `MAP` to `test/exports.js` so you don't have to play tricks with `TreeTransformer`.

Reply:
I'll do that, but I still think that the `skip` functionality should be part of the public API.

Reply:
Hmm, which is `TreeTransformer` in the public API again? 🤔

I mean, CLI only uses `TreeWalker` AFAICT, so we don't need the other one exposed at all.

Reply:
TreeTransformer has always been in the public API and was documented by mishoo on his website.

https://github.com/mishoo/UglifyJS2/blob/df506439b16c550f7e8da30b392f17b0ca4c2e4a/test/exports.js#L13

I've made the changes you've requested in any event.

Reply:
Groan... wrong link above - here's the correct link:

https://github.com/mishoo/UglifyJS2/blob/df506439b16c550f7e8da30b392f17b0ca4c2e4a/tools/exports.js#L5

Reply:
Ah okay, it's for those who wants to manipulate AST directly. Similar reason why `parse` was exported rather than using `minify(code, { output: { ast: true } })`.

In that case, I'll put together a separate PR to export `MAP`  to the public.

Reply:
👍